### PR TITLE
[expense] Created missing functionality for edit + more

### LINKED ISSUE #162: [expense] Create missing functionality for edit/update
**Expense** entity **lacks edit functionality**.
There isn't PUT or PATCH endpoint.
It needs to be added so we can edit its records at the admin panel when needed.

### PR DESCRIPTION
**Pull request by dpmarkov**
1. **Created endpoint** (*PATCH*) for editing an Expense record.
2. **Restricted access** to create/delete/delete all queries. It should be for authorized users only.

close #162

### CODE REVIEW

File: apps/api/src/expenses/expenses.controller.ts [hunk 1]
```diff
@@ -14,9 +25,17 @@ export class ExpensesController {
...
-  create(@Body() createExpenseDto: CreateExpenseDto) {
+  create(
+    @AuthenticatedUser()
+    user: KeycloakTokenParsed,
+    @Body()
+    createExpenseDto: CreateExpenseDto,
+  ) {
+    if (!user) {
+      throw new UnauthorizedException()
+    }
```
Comment (kachar) [inline apps/api/src/expenses/expenses.controller.ts:37]: That's a bit too verbose if we don't actually use the user.

You might as well just skip the `@Public` decorator to require authenticated user. (the default behavior)

An important check here we should add is to make sure only admins can actually update and delete expenses.

Right now any authenticated user can update or delete any expense. That's actually quite important to be fixed @dpmarkov

**Review by kachar — CHANGES_REQUESTED** [review]

File: apps/api/src/expenses/expenses.controller.ts (hunk 1 above)
Comment (dpmarkov) [reply apps/api/src/expenses/expenses.controller.ts:37]: I see, just to be sure does that snippet represent the admin roles?
```javascript
@Roles({
  roles: [RealmViewSupporters.role, ViewSupporters.role],
  mode: RoleMatchingMode.ANY,
})
```

Comment (kachar) [reply apps/api/src/expenses/expenses.controller.ts:37]: Yes, it's exactly what we need here

Comment (dpmarkov) [reply apps/api/src/expenses/expenses.controller.ts]: Sorry for the delay. It's done!

**Review by kachar — APPROVED** [review]

**Pull request merged**